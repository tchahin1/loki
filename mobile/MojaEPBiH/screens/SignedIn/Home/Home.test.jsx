import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import HomeScreen from '.';
import serverTextReducer from '../../../reducers/ServerTextReducer';

const store = createStore(serverTextReducer);

describe('Home snapshot', () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it('renders the home screen', async () => {
    const tree = renderer.create(
      <Provider store={store}>
        <HomeScreen />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
