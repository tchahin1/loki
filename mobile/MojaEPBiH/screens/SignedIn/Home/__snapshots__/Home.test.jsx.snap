// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot renders the home screen 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "paddingTop": 30,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginHorizontal": 50,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgba(96,100,109, 1)",
            "fontSize": 17,
            "lineHeight": 24,
            "textAlign": "center",
          }
        }
      >
        Server response:
         
        Waiting for the response ...
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginTop": 15,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#2e78b7",
              "fontSize": 14,
            }
          }
        >
          Try again if not reachable
        </Text>
      </View>
    </View>
  </View>
</View>
`;
